bundles/org.openhab.persistence.jdbc/src/main/java/org/openhab/persistence/jdbc/internal/JdbcMapper.java

```diff
@@ -252,47 +269,66 @@ protected boolean checkDBAccessability() {
      * DATABASE TABLEHANDLING *
      **************************/
     protected void checkDBSchema() {
-        // Create Items Table if does not exist
-        createItemsTableIfNot(new ItemsVO());
+        if (!conf.getTableUseRealCaseSensitiveItemNames()) {
+            createItemsTableIfNot(new ItemsVO());
+        }
         if (conf.getRebuildTableNames()) {
             formatTableNames();
+
+            if (conf.getTableUseRealCaseSensitiveItemNames()) {
+                dropItemsTableIfExists(new ItemsVO());
+            }
             logger.info(
                     "JDBC::checkDBSchema: Rebuild complete, configure the 'rebuildTableNames' setting to 'false' to stop rebuilds on startup");
-        } else {
             // Reset the error counter
             errCnt = 0;
+        }
+        populateItemNameToTableNameMap();
+    }
+
+    private void populateItemNameToTableNameMap() {
+        itemNameToTableNameMap.clear();
+        if (conf.getTableUseRealCaseSensitiveItemNames()) {
+            for (String itemName : getItemTables().stream().map(t -> t.getTableName()).collect(Collectors.toList())) {
+                itemNameToTableNameMap.put(itemName, itemName);
+            }
+        } else {
             for (ItemsVO vo : getItemIDTableNames()) {
-                sqlTables.put(vo.getItemname(), getTableName(vo.getItemid(), vo.getItemname()));
+                itemNameToTableNameMap.put(vo.getItemName(),
+                        namingStrategy.getTableName(vo.getItemId(), vo.getItemName()));
             }
         }
     }
 
     protected String getTable(Item item) {
-        int rowId = 0;
+        int itemId = 0;
         ItemsVO isvo;
         ItemVO ivo;
 
         String itemName = item.getName();
-        String tableName = sqlTables.get(itemName);
+        String tableName = itemNameToTableNameMap.get(itemName);
 
         // Table already exists - return the name
-        if (tableName != null) {
+        if (!Objects.isNull(tableName)) {
```

Comment:
For my curiosity, what is the difference ?

Reply:
No difference, but in some scenarios can prevent null annotation issues when part of the code is not annotated:

![image](https://user-images.githubusercontent.com/19519842/200061760-303e24d8-62a2-4a9e-9f46-7ef5d2a7a56f.png)

See also #13038, specifically 73fc6c7cec035dee94bc1a9cce597bd6276729e1 from this [comment](https://github.com/openhab/openhab-addons/pull/13038#discussion_r990834520).